code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -152,13 +156,25 @@
 	var/temperature_difference = 0
 	var/skip_tick = TRUE
 	if(!use_enviroment_heat && main_port.total_moles() > 0.01)
-		if(cooling && thermal_exchange_port.total_moles() > 0.01 && nodes[2])
+		if(cooling && thermal_exchange_port.total_moles() > 0.01 && nodes[2] && (thermal_exchange_port.temperature <= FUSION_MAXIMUM_TEMPERATURE || !safeties))
 			thermal_exchange_port.temperature = max(thermal_exchange_port.temperature + heat_amount / thermal_heat_capacity + motor_heat / thermal_heat_capacity, TCMB)
 		else if(cooling && (!thermal_exchange_port.total_moles() || !nodes[2]))
 			skipping_work = skip_tick
 			update_appearance()
 			update_parents()
 			return
+		if(thermal_exchange_port.temperature > FUSION_MAXIMUM_TEMPERATURE && safeties)
+			on = FALSE
+			visible_message("<span class='warning'>The thermal exchange port's temperature has reached critical levels, shutting down...</span>")
+			update_appearance()
+			return
+		else if(thermal_exchange_port.temperature > FUSION_MAXIMUM_TEMPERATURE && !safeties)
+			if((REALTIMEOFDAY - lastwarning) / 5 >= WARNING_DELAY)
+				lastwarning = REALTIMEOFDAY
+				visible_message("<span class='warning'>The thermal exchange port's temperature has reached critical levels!</span>")
```

Comment:
why have safeties in the first place? memes?

Reply:
Is to not have just a barrier that prevents it from going over the limit in a boring way (similarly to how I'm handling walls and HE pipes, instead of making them not work on walls, I made the walls melt)